### PR TITLE
Frustum bounding boxes

### PR DESCRIPTION
**Pull request by Dedado3000**

### CODE REVIEW

File: Source/DataModels/Components/ComponentBoundingBoxes.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "ComponentTransform.h"
+#include "GameObject/GameObject.h"
+
+#include "Application.h"
+#include "ModuleDebugDraw.h"
+
+ComponentBoundingBoxes::ComponentBoundingBoxes(bool active, GameObject* owner)
+	: Component(ComponentType::BOUNDINGBOX, active, owner)
+{
+	localAABB = { {-1, -1, -1}, {1, 1, 1} };
```
Comment (xavi1111) [inline Source/DataModels/Components/ComponentBoundingBoxes.cpp:12]: We will have to remember to change this initial value

File: Source/DataModels/Components/ComponentBoundingBoxes.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	drawBoundingBoxes = false;
+}
+
+void ComponentBoundingBoxes::CalculateBoundingBoxes() 
+{
+	ComponentTransform* transform = (ComponentTransform*)((Component*)this)->GetOwner()->GetComponent(ComponentType::TRANSFORM);
+	objectOBB = OBB(localAABB);
+	objectOBB.Transform(transform->GetGlobalMatrix());
+	encapsuledAABB = objectOBB.MinimalEnclosingAABB();
+}
```
Comment (OriolSors) [inline Source/DataModels/Components/ComponentBoundingBoxes.cpp:24]: We need to revisit this and understand the behaviour

**Review by OriolSors — APPROVED** [review]
Reviewed on discord. @Dedado3000 good fixes.

**Review by xavi1111 — APPROVED** [review]
Good implementation.

**Pull request merged**